### DISCUSSION TITLE
Error when trying to run pegs_calc on other PGS - Catalogs

### DISCUSSION THREAD

**Discussion by PatrickReck01** [PGScatalog/pgsc_calc#220] [Q&A]
Hey,

I am quite new to PGS Scores and I ran into some issues. 

I keep getting the following error:

<img width="1212" alt="Screenshot 2023-12-07 at 15 08 34" src="https://github.com/PGScatalog/pgsc_calc/assets/134544998/a3c60feb-a70f-4afe-9361-b59ec89bb7f0">

<img width="1266" alt="Screenshot 2023-12-07 at 15 07 35" src="https://github.com/PGScatalog/pgsc_calc/assets/134544998/634d83dc-5abb-45c9-bd36-b08bcd138d51">

I get this error when I am trying to use the tool for a vcf file, other than the Test - Sample - Data (the one named 'cineca').
I downloaded the vcf file from 1000Genomes: https://www.internationalgenome.org/data-portal/sample/SC_GMJOL5309922

I also tried it with a dataset I got from Kaggle and transformed into a vcf file, however I got the same error. The Kaggle dataset is this from here: https://www.kaggle.com/datasets/zusmani/mygenome

Interestingly enough, even for the Test Dataset to work, I couldn't use the --pgs_id PGS001229 argument. Instead, I  had to use the argument --scorefile '+scoringfilepath+' with scoringfilepath being the path to the scoring file.txt that I found in the folder:
/Users/patrickreck/.nextflow/assets/pgscatalog/pgsc_calc/assets/examples/scorefiles/scorefile.txt

If anyone has any ideas hot to fix this, I would be very thankful. I have been trying to figure this out for a while now, but I don't seem to get any closer.

Best regards
Patrick

**Comment by nebfield** [comment]
This problem happens when none of the genetic variants in the scoring file can be found in your target (input) genomes. It's typically a problem with the input genotypes or the selected parameters:

* did you accidentally select the wrong genome build?
  *  we match variants by looking at genomic coordinates, which change across different genome builds
* do your input genomes have [low variant density](https://en.wikipedia.org/wiki/Imputation_(genetics))?
  * [we recommend imputing target genomes before using the calculator](https://pgsc-calc.readthedocs.io/en/latest/how-to/prepare.html#target-genome-data-requirements)

If you're looking for some publicly availble data to learn about PGS I suggest the 1000 Genomes dataset, [which is helpfully hosted by the plink2 developers](https://www.cog-genomics.org/plink/2.0/resources). These genotypes have good density and should work with many scores. [HAPNEST is also nice, but it's synthetic data and very large](https://www.ebi.ac.uk/biostudies/studies/S-BSST936).

**Reply by PatrickReck01** [reply]
Thanks you so much for the fast reply.

I checked, the genome build for both the scoring file and the vcf file I was using and they both seem to be GRCh37 / hg 19.

As for imputing the target genomes: I didn't do it and I am not sure what it does,  however I am gonna try it right away. 

Also thanks for the links to the extra datasets, I am gonna try them as soon as I can. 

I hope it is ok if I leave this thread open, in case I have any more questions that come up? 
I am really starting to worry about the tool not working, since I need it for my bachelor thesis.

**Reply by PatrickReck01** [reply]
Another question that I just thought of:

In the documentation for pgs_calc it says, that the input data should consist of multiple samples, however I am trying to build a web application, where a user can upload his genomic data and receive his pgs scores. That would mean that the input data would only consist of one sample from one individual. Is there any mayor problem with that, or would it work as well?

**Reply by nebfield** [reply]
Downloading the datasets I mentioned is probably easier than trying to impute the data you have. Genotype imputation can be complicated and confusing.

That's an exciting project! There may be some challenges with low sample sizes. Typically people recommend at least  50 to 100 samples in the target (input) sample. See:

https://www.nature.com/articles/s41596-020-0353-1#Fig2
https://choishingwan.github.io/PRS-Tutorial/target/#sample-size 

If you can run `pgsc_calc` with `--run_ancestry` low sample sizes are supported because we load missing data from reference populations. However, the [ancestry normalisation](https://pgsc-calc.readthedocs.io/en/latest/how-to/ancestry.html) might be too complicated to include in your project. Good luck!

**Reply by PatrickReck01** [reply]
I tried to run the datasets (https://www.cog-genomics.org/plink/2.0/resources#phase3_1kg  - 1000Genomes phase 3 - GrCh 37) from above, however now I am getting the following error:

![Screenshot 2023-12-10 at 19 36 47](https://github.com/PGScatalog/pgsc_calc/assets/134544998/8a5b644a-f043-481c-8803-079455dc3fef)
![Screenshot 2023-12-10 at 19 37 02](https://github.com/PGScatalog/pgsc_calc/assets/134544998/d53afc6b-7ce2-46c3-a31d-dc476c193504)
![Screenshot 2023-12-10 at 19 37 47](https://github.com/PGScatalog/pgsc_calc/assets/134544998/8991bb6e-4bd8-42b8-88be-37b5a47dc61e)
![Screenshot 2023-12-10 at 19 38 11](https://github.com/PGScatalog/pgsc_calc/assets/134544998/225b89dc-a7b6-49ca-8ebd-8a9255c40185)

Do you have any idea what could cause this? I am not really getting any information out of the error message itself.

As an alternative, I read a forum post (https://www.biostars.org/p/9477182/ ) and it got me thinking, whether I could simply calculate a PGS 'by hand' by calculating the sum over all the SNPs that appear in both the respective pgs-Catalog and the input genome file, weighted by their effect weight. I am assuming, the quality of the PRS score I get this way is lower, however is this option at least somewhat viable?

Alternatively, would it be possible to add my input sample from 1 individual to a bigger dataset with samples from many individuals and use this to calculate the PGS scores? I could then go into results, search the sample I added and read its PGS score.

I also wanted to ask, why there are always datasets with samples from multiple individuals are needed? I read in a paper (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7612115/ ) its because of quality control mechanisms, however I don't really understand why other individuals genome data would influence the PGS of my single input genome file from 1 individual. I get that large sample sizes is important in GWAS when you are determining which SNPs have an influence on a phenotype, but why is it important when calculating single PGS scores?

I hope I am not bothering you with all my questions. :)
Thanks in advance 👍

**Reply by hereagain-Y** [reply]
Hi Patrick, when I was trying to use the vcf data from I ran into the same issue as you posted previously, I was wondering did you have any clue to fix it?

**Reply by smlmbrt** [reply]
@hereagain-Y - what is the exact problem you're having? That no variants match?
